### PR TITLE
Bnb/ddh norm

### PR DESCRIPTION
**Pull request by bnb32**

### CODE REVIEW

File: sup3r/preprocessing/batch_handling.py [hunk 1]
```diff
@@ -618,14 +618,15 @@ def _parallel_normalization(self):
...
+                dh.normalize(self.means, self.stds,
+                             max_workers=dh.norm_workers)
         else:
             with ThreadPoolExecutor(max_workers=max_workers) as exe:
                 futures = {}
                 now = dt.now()
                 for idh, dh in enumerate(self.data_handlers):
                     future = exe.submit(dh.normalize, self.means, self.stds,
-                                        max_workers=1)
+                                        max_workers=dh.norm_workers)
```
Comment (grantbuster) [inline sup3r/preprocessing/batch_handling.py:629]: Is there a specific reason you would want a worker to spin up more threads within a thread pool? Historically i've been very cautious of this. I don't want my threads/processes to be forking themselves.

File: sup3r/preprocessing/data_handling/base.py [hunk 2]
```diff
@@ -1269,14 +1200,12 @@ def serial_data_fill(self, shifted_time_chunks):
...
-            if t % interval == 0:
-                logger.info(f'Added {t + 1} of {len(shifted_time_chunks)} '
-                            'chunks to final data array')
+                self._data_fill(t, ts, f_index, f)
+            logger.info(f'Added {t + 1} of {len(shifted_time_chunks)} '
+                        'chunks to final data array')
             self._raw_data.pop(t)
 
-    def parallel_data_fill(self, shifted_time_chunks, max_workers=None):
+    def data_fill(self, shifted_time_chunks, max_workers=None):
```
Comment (grantbuster) [inline sup3r/preprocessing/data_handling/base.py:1208]: Not a huge fan of this naming convention. I always find it confusing when there is a `.method` and a `._method`. I think you're trying to hide the serial method from the "public" view but it makes the developer's job more confusing. I'd suggest `_single_data_fill()` and `_data_fill()`

**Review by grantbuster — CHANGES_REQUESTED** [review]
Might have a few negative tests fail that i added recently, i can handle them if you want.

File: sup3r/preprocessing/batch_handling.py (hunk 1 above)
Comment (bnb32) [reply sup3r/preprocessing/batch_handling.py:629]: Yeah that's fair. Makes sense to go back to 1

**Review by grantbuster — APPROVED** [review]

**Pull request merged**